cmd/uzfs_test/zrepl_utest.c

```diff
@@ -0,0 +1,411 @@
+#include <stdio.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <strings.h>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+#include <netdb.h>
+#include <unistd.h>
+#include <pthread.h>
+#include <string.h>
+#include <zrepl_prot.h>
+#include <sys/zfs_context.h>
+#include <uzfs_test.h>
+#include <uzfs_mgmt.h>
+
+char *accpt_port = "3232";
+char *tgt_port = "6060";
+
+struct data_io {
+	zvol_io_hdr_t hdr;
+	char buf[0];
+};
+
+static int
+create_and_bind(char *port)
+{
+	int s, sfd;
+	struct addrinfo hints = {0, };
+	struct addrinfo *result, *rp;
+
+	hints.ai_family = AF_INET;
+	hints.ai_socktype = SOCK_STREAM;
+	hints.ai_flags = AI_PASSIVE;
+
+	s = getaddrinfo(NULL, port, &hints, &result);
+	if (s != 0) {
+		printf("getaddrinfo failed with error\n");
+		return (-1);
+	}
+
+	for (rp = result; rp != NULL; rp = rp->ai_next) {
+		sfd = socket(rp->ai_family, rp->ai_socktype, rp->ai_protocol);
+		if (sfd == -1) {
+			continue;
+		}
+
+		s = bind(sfd, rp->ai_addr, rp->ai_addrlen);
+		if (s == 0) {
+			/* We managed to bind successfully! */
+			printf("bind is successful\n");
+			break;
+		}
+
+		close(sfd);
+	}
+
+	if (rp == NULL) {
+		printf("bind failed with err\n");
+		return (-1);
+	}
+
+	freeaddrinfo(result);
+	return (sfd);
+}
+
+void
+populate(char *p, int size)
+{
+
+	int i;
+
+	for (i = 0; i < size; i++) {
+		p[i] = 'C';
+	}
+}
+
+int
+zrepl_verify_data(char *p, int size)
+{
+
+	int i;
+
+	for (i = 0; i < size; i++) {
+		if (p[i] != 'C') {
```

Comment:
using random data would be good

Reply:
How we will verify ? Storing in temp file ?? I seen that in existing verify code, it checks non zero char at selective byte locations in buffer. I think more deteministic way is that you know what data you have written and what you are getting.

Reply:
we can have in-memory.. lets say of 1G volsize, so, 1G of in-memory

Reply:
Just curious to know what purpose it will serve by having random data written over wire instead of stashing some know string/data and sending it over wire ?

Reply:
data being same, we won't be able to test any corruption or data overwrites due to synchronization issues at (different layers of) zfs